packages/functional-tests/tests/syncV3/syncV3ResetPassword.spec.ts

```diff
@@ -0,0 +1,102 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { test, expect } from '../../lib/fixtures/standard';
+import { EmailHeader, EmailType } from '../../lib/email';
+
+const PASSWORD = 'passwordzxcv';
+
+test.describe('Firefox Desktop Sync v3 reset password', () => {
+  test.beforeEach(({}, testInfo) => {
+    test.slow(testInfo.project.name !== 'local', 'email delivery can be slow');
+  });
+
+  test('reset password, verify same browser', async ({
+    target,
+    page,
+    pages: { login, resetPassword },
+  }) => {
+    const uaStrings = {
+      desktop_firefox_58:
+        'Mozilla/5.0 (Windows NT 6.1; WOW64; rv:58.0) Gecko/20100101 Firefox/58.0',
+    };
+    const query = { forceUA: uaStrings['desktop_firefox_58'] };
+    const email = login.createEmail();
+    await target.auth.signUp(email, PASSWORD, {
+      lang: 'en',
+      preVerified: 'true',
+    });
+    await login.clearCache();
+    await page.goto(
+      `${
+        target.contentServerUrl
+      }/reset_password?context=fx_desktop_v3&service=sync/?${query.toString()}`
+    );
+    await resetPassword.fillOutResetPassword(email);
+
+    const link = await target.email.waitForEmail(
+      email,
+      EmailType.recovery,
+      EmailHeader.link
+    );
+    await page.goto(link, { waitUntil: 'load' });
+    await resetPassword.resetNewPassword('Newpassword@');
+    expect(await login.loginHeader()).toBe(true);
+  });
+
+  test('reset password, verify same browser, password validation', async ({
+    target,
+    page,
+    pages: { login, resetPassword },
+  }) => {
+    const query = {
+      forceExperiment: 'passwordStrength',
+      forceExperimentGroup: 'designF',
+    };
+    const email = login.createEmail();
+    await target.auth.signUp(email, PASSWORD, {
+      lang: 'en',
+      preVerified: 'true',
+    });
+    await login.clearCache();
+    await page.goto(
+      `${target.contentServerUrl}/reset_password?context=fx_desktop_v3&service=sync/?`
```

Comment:
This URL feels a bit funny with the `/?` at the end of the service name query param.